compiler/rustc_resolve/src/ident.rs

```diff
@@ -900,7 +895,7 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             }
 
             // Forbid expanded shadowing to avoid time travel.
-            if let Some(shadowed_glob) = resolution.shadowed_glob
+            if let Some(shadowed_glob) = resolution.glob_binding()
```

Comment:
This line alone looks like a change in semantics - the binding is still glob now, but not necessarily shadowed.
But the `binding.res() != shadowed_glob.res()` condition below should ensure the old behavior, right?